model/riscv_step.sail

```diff
@@ -8,8 +8,117 @@
 
 /* The emulator fetch-execute-interrupt dispatch loop. */
 
-/* returns whether to increment the step count in the trace */
-function step(step_no : int) -> bool = {
+register step_state : StepState
+
+function init_stepper() -> unit = {
+  step_state = STEP_ACTIVE;
```

Comment:
I think you could just initialise this like

```
register step_state : StepState = STEP_ACTIVE
```

Reply:
I could, but `init_stepper` is also needed for RVFI, since it initializes the model multiple times (C:`reinit_sail()` -> Sail:`init_model()` -> Sail:`init_stepper()`).  So I just left it in `init_stepper()`.

Reply:
RVFI calls the generated C code that reruns the global initialisers (`model_init()` IIRC), so it's safe to do it using that.